fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -4717,7 +4717,13 @@ public ParseNode visitOdbcFunctionCallExpression(StarRocksParser.OdbcFunctionCal
     private static List<Expr> getArgumentsForTimeSlice(Expr time, Expr value, String ident, String boundary) {
         List<Expr> exprs = Lists.newLinkedList();
         exprs.add(time);
-        exprs.add(value);
+        // IntLiteral may generate TINYINT/SMALLINT/INT/LARGEINT
+        // but time_slice only support INT type when executed in BE
+        if (value instanceof IntLiteral) {
```

Comment:
You‘d better test this is a const value, then test if it can be computed in Frontend.
And if the value can be cast to an intvalue.
Like time_slice(""xxxx, 1 + 1);

Reply:
Already check in analyzer
```
            } else if (fnName.equals(FunctionSet.TIME_SLICE) || fnName.equals(FunctionSet.DATE_SLICE)) {
                // This must before test for DecimalV3.
                if (!(node.getChild(1) instanceof IntLiteral)) {
                    throw new SemanticException(
                            fnName + " requires second parameter must be a constant interval");
                }
                if (((IntLiteral) node.getChild(1)).getValue() <= 0) {
                    throw new SemanticException(
                            fnName + " requires second parameter must be greater than 0");
                }
```